SocialCareCaseViewerApi.Tests/V1/Gateways/DatabaseGateway.Tests.cs

```diff
@@ -2330,6 +2330,47 @@ public void UpdatePersonSetsCorrectValuesForPersonEntity()
             person.ReviewDate.Should().Be(request.ReviewDate);
         }
 
+        [Test]
+        public void UpdatePersonsAddressUpdatesTheMostRecentAddressToDisplay()
+        {
+            var first = DateTime.Today;
+            var second = DateTime.Today.AddDays(-1);
+            var last = DateTime.Today.AddDays(-2);
+
+            var person = DatabaseGatewayHelper.CreatePersonDatabaseEntity();
+            person.Addresses = null;
+            var firstAddress = DatabaseGatewayHelper.CreateAddressDatabaseEntity(person.Id, startDate: first, isDisplayAddress: "Y");
+            var secondAddress = DatabaseGatewayHelper.CreateAddressDatabaseEntity(person.Id, startDate: second, isDisplayAddress: "Y");
+            var lastAddress = DatabaseGatewayHelper.CreateAddressDatabaseEntity(person.Id, startDate: last, isDisplayAddress: "Y");
+
+            DatabaseContext.Persons.Add(person);
+            DatabaseContext.Addresses.Add(firstAddress);
+            DatabaseContext.Addresses.Add(secondAddress);
+            DatabaseContext.Addresses.Add(lastAddress);
+            DatabaseContext.SaveChanges();
+
+            var personBeforeUpdate = DatabaseContext.Persons.Find(person.Id);
+            var addressBeforeUpdate = DatabaseContext.Addresses.FirstOrDefault(x => x.PostCode == firstAddress.PostCode && x.StartDate == first && x.PersonId == person.Id);
+
+            personBeforeUpdate.Addresses.Count.Should().Equals(3);
+            addressBeforeUpdate.IsDisplayAddress.Should().Equals("Y");
+
+            var request = GetValidUpdatePersonRequest(person.Id);
+
+            _classUnderTest.UpdatePerson(request);
+
+            var updatedPerson = DatabaseContext.Persons.Find(person.Id);
+            var previousAddress = DatabaseContext.Addresses.FirstOrDefault(x => x.PostCode == firstAddress.PostCode && x.StartDate == first && x.PersonId == person.Id);
+            var newAddress = DatabaseContext.Addresses.FirstOrDefault(x => x.PostCode == request.Address.Postcode && x.PersonId == person.Id);
```

Comment:
If you have the original IDs in `existingAddresses` list, you can grab the new address by id by excluding the old ones, like `var newAddress = DatabaseContext.Addresses.FirstOrDefault(x => !existingAddresses.Select(x => x.AddressId).Contains(x.AddressId));`

That should make everything ID based so the values don't matter at all.